generator/autorest.ts

```diff
@@ -2,15 +2,97 @@
 // Licensed under the MIT License.
 import path from 'path';
 import os from 'os';
-import { findRecursive, lowerCaseContains, executeCmd, fileExists } from './utils';
+import { findRecursive, executeCmd, fileExists } from './utils';
 import * as constants from './constants';
-import { ReadmeTag, AutoGenConfig, CodeBlock } from './models';
-import * as cm from '@ts-common/commonmark-to-markdown'
-import * as yaml from 'js-yaml'
 import { readFile, writeFile } from 'fs/promises';
+import * as markdown from '@ts-common/commonmark-to-markdown'
+import * as yaml from 'js-yaml'
 
 const autorestBinary = os.platform() === 'win32' ? 'autorest.cmd' : 'autorest';
-export const apiVersionRegex = /^\d{4}-\d{2}-\d{2}(|-preview)$/;
+
+const rootDir = `${__dirname}/../`;
+const extensionDir = path.resolve(`${rootDir}/bicep-types-az/src/autorest.bicep/`);
+
+export async function generateAutorestConfig(readmePath: string, bicepReadmePath: string) {
+  // We expect a path format convention of <provider>/(any/number/of/intervening/folders)/<yyyy>-<mm>-<dd>(|-preview)/<filename>.json
+  // This information is used to generate individual tags in the generated autorest configuration
+  // eslint-disable-next-line no-useless-escape
+  const pathRegex = /^(\$\(this-folder\)\/|)([^\/]+)(?:\/[^\/]+)*\/(\d{4}-\d{2}-\d{2}(|-preview))\/.*\.json$/i;
+
+  const readmeContents = await readFile(readmePath, { encoding: 'utf8' });
+  const readmeMarkdown = markdown.parse(readmeContents);
+
+  const inputFiles = new Set<string>();
+  // we need to look for all autorest configuration elements containing input files, and collect that list of files. These will look like (e.g.):
+  // ```yaml $(tag) == 'someTag'
+  // input-file:
+  // - path/to/file.json
+  // - path/to/other_file.json
+  // ```
+  for (const node of markdown.iterate(readmeMarkdown.markDown)) {
+    // We're only interested in yaml code blocks
+    if (node.type !== 'code_block' || !node.info || !node.literal ||
+      !node.info.trim().startsWith('yaml')) {
+      continue;
+    }
```

Comment:
This section guarantees that we only read, right?: 

``` yaml $(tag) == 'package-bicep-2023-11'
input-file:
  - Microsoft.Resources/stable/2023-11-01/bicepClient.json
```

Reply:
Yes, exactly.